src/sfError.ts

```diff
@@ -130,24 +165,13 @@ export class SfError<T = unknown> extends NamedError {
    * Convert an {@link SfError} state to an object. Returns a plain object representing the state of this error.
    */
   public toObject(): JsonMap {
-    const obj: JsonMap = {
+    return {
       name: this.name,
       message: this.message ?? this.name,
       exitCode: this.exitCode,
       actions: this.actions,
+      ...(this.context ? { context: this.context } : {}),
+      ...(this.data ? { data: this.data } : {}),
```

Comment:
We still can't type this better than `JsonMap`?

Reply:
oh, we can!